src/FakeItEasy/ExceptionThrowerConfigurationExtensions.cs

```diff
@@ -67,11 +70,12 @@ public static IAfterCallSpecifiedConfiguration Throws<T1>(this IExceptionThrower
         /// </summary>
         /// <param name="configuration">The configuration to use.</param>
         /// <param name="exceptionFactory">A function that returns the exception to throw when invoked.</param>
+        /// <typeparam name="TInterface">The type of configuration interface to return.</typeparam>
         /// <typeparam name="T1">Type of the first argument of the faked method call.</typeparam>
         /// <typeparam name="T2">Type of the second argument of the faked method call.</typeparam>
         /// <returns>Configuration object.</returns>
         /// <exception cref="FakeConfigurationException">The signatures of the faked method and the <paramref name="exceptionFactory"/> do not match.</exception>
-        public static IAfterCallSpecifiedConfiguration Throws<T1, T2>(this IExceptionThrowerConfiguration configuration, Func<T1, T2, Exception> exceptionFactory)
+        internal static IAfterCallSpecifiedConfiguration<TInterface> Throws<TInterface, T1, T2>(this IExceptionThrowerConfiguration<TInterface> configuration, Func<T1, T2, Exception> exceptionFactory)

```

Comment:
The addition of the `TInterface` type param made this method, and the following ones, too impractical to use, so they had to be moved to the `IExceptionThrowerConfiguration<TInterface>`. The extension methods are still used by the classes that implement the interface, to avoid code duplication.